File: backend/app/models/entry.rb
class Entry < ApplicationRecord
  
  belongs_to :jam
  belongs_to :user

  has_many :vote_tokens
  # TODO: add unique user_id in context of jam_id


  # View
  def artist_name
    self.user.username
  end

  def votes
    self.vote_tokens.pluck(:user_id)
  end
end

File: frontend/src/types/Entry.ts
import { VoteToken } from 'types/VoteToken'

export type Entry = {
  id: string
  link: string
  title: string
  user_id: string // FK
  jam_id: string // FK
  created_at?: string
  updated_at?: string
}

export type EntryView = Entry & {
  // entry.votes
  artist_name: string
  votes?: VoteToken[]
}
